dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/utils/DagHelper.java

```diff
@@ -423,7 +423,7 @@ private static List<String> skipTaskNode4Switch(TaskNode taskNode, Map<String, T
         if (CollectionUtils.isEmpty(switchTaskList)) {
             switchTaskList = new ArrayList<>();
         }
-        conditionResultVoList.remove(resultConditionLocation);
+        // conditionResultVoList.remove(resultConditionLocation);
```

Comment:
Why remove this code?

Reply:
Because I find we will remove the location here and this will cause the default branch cannot reach.